### PR TITLE
Fundamentals TOC

### LINKED ISSUE #20342: Move Standard and Core docs to new Fundamentals TOC
* Replace .NET guide TOC and .NET Core guide TOC with .NET fundamentals TOC. 
* Replace TOC nodes .NET guide and .NET Core guide in the [top-level TOC](https://docs.microsoft.com/en-us/dotnet/welcome) with .NET Fundamentals.
* Replace .NET guide and .NET Core guide landing pages with a new .NET fundamentals landing page.

Exclude from this initial PR as impractical to do all at one blow, will be addressed piecemeal using separate issues later:

* File name changes, URL redirects.
* Creation of new docs, changes within docs.
* Updating “Core” terminology in TOC text.

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #20342

### CODE REVIEW

**Review by gewarren — COMMENTED** [review]
This looks really good Tom! Nice job taking on this initial work.

File: docs/fundamentals/toc.yml [hunk 1]
```diff
@@ -0,0 +1,1330 @@
...
+      - name: Analyzers
+        items:
+        - name: Overview
+          href: ../standard/analyzers/index.md
+          displayName: analyzers
+        - name: API analyzer
+          href: ../standard/analyzers/api-analyzer.md
+        - name: Portability analyzer
+          href: ../standard/analyzers/portability-analyzer.md
+        - name: Framework analyzer
```
[the remark this answers is not in the document]
Comment (tdykstra) [reply docs/fundamentals/toc.yml:349]: >This analyzer is specific to .NET Framework.

I'll move it to the Framework TOC

**Comment by tdykstra** [comment]
Thanks for the review @gewarren, great catches!

**Review by gewarren — APPROVED** [review]

File: docs/architecture/microservices/net-core-net-framework-containers/net-framework-container-scenarios.md [hunk 2]
```diff
@@ -43,8 +43,8 @@ In the meantime, if any platform or service in Azure still doesn't support .NET
 
 ### Additional resources
 
-- **.NET Core Guide** \
-  [https://docs.microsoft.com/dotnet/core/index](../../../core/index.yml)
+- **.NET fundamentals** \
+  [https://docs.microsoft.com/dotnet/core/index](../../../fundamentals/index.yml)
```
Comment (Youssef1313) [inline docs/architecture/microservices/net-core-net-framework-containers/net-framework-container-scenarios.md:47]: Link text needs to be changed too.
I also can't find index.yml in fundamentals folder. However, the build isn't reporting a warning for it. Am I missing something?

File: docs/fundamentals/toc.yml [hunk 3]
```diff
@@ -0,0 +1,1323 @@
...
+  - name: Choose between anonymous and tuple types
+    href: ../standard/base-types/choosing-between-anonymous-and-tuple.md
+- name: Microsoft Extensions 
+  items:
+  - name: Configuration
+    href: /aspnet/core/fundamentals/configuration/
+  - name: Dependency injection
+    href: /aspnet/core/fundamentals/dependency-injection/
+  - name: Logging
+    href: /aspnet/core/fundamentals/logging/
```
Comment (IEvangelist) [inline docs/fundamentals/toc.yml:698]: These need to be ported over from https://github.com/dotnet/AspNetCore.Docs to our docs - right? I'll create an issue to capture that effort.

File: docs/fundamentals/toc.yml [hunk 4]
```diff
@@ -0,0 +1,1323 @@
...
+        href: ../standard/parallel-programming/introduction-to-plinq.md
+      - name: Understanding Speedup in PLINQ
+        href: ../standard/parallel-programming/understanding-speedup-in-plinq.md
+      - name: Order Preservation in PLINQ
+        href: ../standard/parallel-programming/order-preservation-in-plinq.md
+      - name: Merge Options in PLINQ
+        href: ../standard/parallel-programming/merge-options-in-plinq.md
+      - name: Potential Pitfalls with PLINQ
+        href: ../standard/parallel-programming/potential-pitfalls-with-plinq.md
+      - name: "How to: Create and Execute a Simple PLINQ Query"
```
Comment (IEvangelist) [inline docs/fundamentals/toc.yml:819]: Now is probably a good time to correct the character casing of all these TOC entries, shouldn't they be sentence case?

**Review by IEvangelist — COMMENTED** [review]
A few questions, but this is AWESOME - great job @tdykstra 👍🏼

File: docs/fundamentals/toc.yml (hunk 3 above)
Comment (tdykstra) [reply docs/fundamentals/toc.yml:698]: We have one already - https://github.com/dotnet/docs/issues/4567

File: docs/fundamentals/toc.yml (hunk 4 above)
Comment (tdykstra) [reply docs/fundamentals/toc.yml:819]: That can be done anytime, there's enough going on in this PR and that's super low priority, so I'd rather leave that for dedicated PRs after this one.  

@Youssef1313 I see you put 👍 on this comment, would you like to submit PR(s) to fix these titles after this PR is merged?

**Comment by tdykstra** [comment]
Thanks for the review, @IEvangelist !

File: docs/fundamentals/toc.yml (hunk 4 above)
Comment (Youssef1313) [reply docs/fundamentals/toc.yml:819]: @tdykstra Sure.

**Pull request merged**